tests/integration/workflows/test_task_merging.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2014-2017 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+from __future__ import absolute_import, division, print_function
+
+import pytest
+
+from invenio_db import db
+
+from inspirehep.modules.records.api import InspireRecord
+from inspirehep.modules.workflows.tasks.merging import (
+    _read_wf_record_source,
+    _insert_wf_record_source
+)
+
+
+@pytest.fixture()
+def dummy_record(workflow_app):
+    record = InspireRecord.create({
+        "$schema": "http://localhost:5000/schemas/records/hep.json",
+        "_collections": ['Literature'],
+        "titles": [{
+            "title": "foo"
+        }],
+        "document_type": ["thesis"]
+    })
+    yield record
+    record._delete(force=True)
+
+
+def test_wf_record_source_read_and_write(dummy_record):
+    record_uuid = dummy_record.id
+    original_root = {"title": "baz"}
+
+    _insert_wf_record_source(
+        json=original_root,
+        record_uuid=record_uuid,
+        source='arXiv'
+    )
+    db.session.commit()
+
+    retrieved_root = _read_wf_record_source(record_uuid=record_uuid, source='arXiv')
+    assert original_root == retrieved_root
+
+
+def test_empty_root(dummy_record):
+    record_uuid = dummy_record.id
+    retrieved_root = _read_wf_record_source(record_uuid=record_uuid, source='Elsevier')
+    assert retrieved_root == {}
```

Comment:
Should it be case sensitive or not? (I vote for not caring about the casing)

Reply:
I would add also the tests where the database has stuff and the source does not match (as the db.session.commit is done out of the fixture, the dummy_record does not really get into db).